core/src/dunit/scala/io/snappydata/cluster/SplitClusterDUnitTest.scala

```diff
@@ -437,12 +440,12 @@ object SplitClusterDUnitTest extends SplitClusterDUnitTestObject {
     // check validateAll and clearing of "validated" flag and reset for failures
 
     // for ARRAY serializers(0)
-    serializers(0).serialize(dec1)
+    serializers(0).serialize(dbl1)
     try {
-      serializers(0).serialize(dec4)
-      Assert.fail("Expected an IllegalArgumentException")
+      serializers(0).serialize(dec4) // String cannot be casted to Double
+      Assert.fail("Expected an ClassCastException")
     } catch {
-      case _: IllegalArgumentException => // expected
+      case _: ClassCastException => // expected
```

Comment:
Can you check why this is throwing ClassCastException? Which is the line that is trying to cast? Normally this should throw an IllegalArgumentException from GenericValidatingConverter.apply